### PR TITLE
Do not open indices with broken index settings

### LINKED ISSUE #26998: Opening indices with broken index settings
Today we are lenient with index settings, we do the following things:
1. We will open an index even if it contains an index setting that we do not recognize because we do not do any index settings validation when opening an index.
2. We will upgrade an index and archive unrecognized settings; this is bad in the following scenario:
  a. a user installs a plugin that provides an index setting
  b. a user creates an index that uses that index setting
  c. a user stops Elasticsearch to upgrade, so removes the plugin
  d. the user upgrades Elasticsearch, and starts Elasticsearch not having reinstalled the plugin
  e. Elasticsearch will archive the index setting and happily open the index

This leniency is dangerous and we should stop.

To be fair, the reason for the existence of this leniency in the upgrade case is for when we first introduced strict settings, we knew that we would see unrecognized settings on indices coming from 2.x into 5.x. We are past that point now. It's time to reconsider this leniency.

Relates #26995

### PR DESCRIPTION
**Pull request by mayya-sharipova**
- remove the leniency on opening indices with unrecognized settings and
instead such an index will be closed with the unrecognized settings archived
- do not open any index with archived settings
- users can remove archived settings via the wildcard archived
- return "index_open_exception" 400 status code when trying to open an
index with broken settings

Relates to #26995
Closes #26998

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/IndexOpenException.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Exception indicating that index can not be opened
+ */
+public class IndexOpenException extends ElasticsearchException {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/indices/IndexOpenException.java]: I do not think we need a new exception type? Is there an existing one that we can reuse?

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 2]
```diff
@@ -434,7 +434,12 @@ private synchronized IndexService createIndexService(final String reason,
                                                          IndexingOperationListener... indexingOperationListeners) throws IOException {
         final IndexSettings idxSettings = new IndexSettings(indexMetaData, this.settings, indexScopedSettings);
         // we ignore private settings since they are not registered settings
-        indexScopedSettings.validate(indexMetaData.getSettings(), true, true, true);
+        if (reason.equals("metadata verification")) {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java]: Instead of a string comparison, can we push a boolean down?

**Review by jasontedor — COMMENTED** [review]
I left some questions.

File: server/src/main/java/org/elasticsearch/indices/IndexOpenException.java (hunk 1 above)
Comment (mayya-sharipova) [reply server/src/main/java/org/elasticsearch/indices/IndexOpenException.java]: @jasontedor Thanks Jason for the review! I could not find any relevant exception. `ILLEGAL_INDEX_SHARD_STATE_EXCEPTION` - 404, `INDEX_CLOSED_EXCEPTION` - 400? `ElasticsearchException` that was used in the code before produces 500, while I think we want to produce 400, as it is a user's error to open an index with broken settings. What would you suggest?

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/indices/IndexOpenException.java]: Why not `IllegalArgumentException`?

**Comment by mayya-sharipova** [comment]
Closing this PR, as it looks like it is stalled. If the issue is still relevant, people are welcome to work on it.

**Pull request closed without merging**